src/Hangman/HangmanGame.java

```diff
@@ -0,0 +1,208 @@
+package Hangman;
+
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Scanner;
+
+public class HangmanGame {
+
+    private String[] board;
+
+    private static String word;
+
+    private static int lengthOfWord;
+
+    private static String[] lettersToGuess;
+
+    private static int incorrectGuesses = 0;
+
+    private static int correctGuesses = 0;
+    private static HashSet<String> lettersGuessed;
+
+
+
+    public HangmanGame(){
+        board = new String[]{"  ┌──┐",
+                             "     │",
+                             "     │",
+                             "     │",
+                             "   ──┴──"};
+        lettersGuessed = new HashSet<String>();
+    }
+
+    public void printBeginningBoard(){
+        System.out.println("  ┌──┐");
+        System.out.println("  O  │      Start game? (y/n)");
+        System.out.println(" /|\\ │");
+        System.out.println(" / \\ │");
+        System.out.println("   ──┴──");
+    }
+    public void printEndBoard(){
+        System.out.println("  ┌──┐");
+        System.out.println("  O  │      Thank you for playing!");
+        System.out.println(" /|\\ │");
+        System.out.println(" / \\ │");
+        System.out.println("   ──┴──");
+    }
+
+    public void startGame(){
+        printBeginningBoard();
+        Scanner input = new Scanner(System.in);
+        String choice = input.nextLine();
+        if(choice.toLowerCase().equals("n")) printEndBoard();
+        else if (choice.toLowerCase().equals("y")) playGame();
+        else {
+            System.out.println("You must pick either Y or N!");
+            startGame();
+        }
+    }
+
+    public void printBoard(){
+        for (String line : board) {
+            System.out.println(line);
+        }
+    }
+
+    public void printLetters(){
+        for (String toGuess : lettersToGuess) {
+            System.out.print(toGuess);
+        }
+        messageToPlayer();
+    }
+
+    public void playGame(){
```

Comment:
I know you said there where many places you know you can improve on it so hopefully i'm not pointing out things you already know. But there are a couple things i see. The code works perfectly and i can understand pretty easily but just a few pointers. 

1.) I think variable names should be a little more descriptive, this is a easy project but just getting into the habit of leaning on the sides of being a little to descriptive with your variables and method names rather than un-descriptive. So for instance instead of a variable called 'word' maybe call it 'randomWord' or instead of a var named 'input' it could be 'inputFromUser'. These seem small but when you have a complex method the variable names makes it kind of read like english. 

2. This is a pretty big method that is doing a lot of things. Ideally you'd want your each method to complete a specific task. In this playGame method it does a couple things, it sets either the production or test mode. It sets the initial playing board. It checks the game status of the game, it filters the users input to make sure its a valid input. It multiple a couple things. Hypothetically if this was a really complex project with complex implementation details this method would be hell since its so many complex moving parts and doing so many things within the method it could be overwhelming. Not even mentioning maintaining or fixing a bug that is in this method. Not a biggie for this small project, but just a good habit to get into